lib/modules/platform/gerrit/scm.ts

```diff
@@ -0,0 +1,158 @@
+      .findChanges(repository, searchConfig)
+      .then((res) => res.pop());
+    if (change) {
+      return true;
+    }
+    return git.branchExists(branchName);
+  }
+
+  override async getBranchCommit(
+    branchName: string
+  ): Promise<CommitSha | null> {
+    const searchConfig: GerritFindPRConfig = { state: 'open', branchName };
+    const change = await client
+      .findChanges(repository, searchConfig)
+      .then((res) => res.pop());
+    if (change) {
+      return change.current_revision!;
+    }
+    return git.getBranchCommit(branchName);
+  }
+
+  override async isBranchBehindBase(
+    branchName: string,
+    baseBranch: string
+  ): Promise<boolean> {
+    const searchConfig: GerritFindPRConfig = {
+      state: 'open',
+      branchName,
+      targetBranch: baseBranch,
+    };
+    const change = await client
+      .findChanges(repository, searchConfig, true)
+      .then((res) => res.pop());
+    if (change) {
+      const currentGerritPatchset = change.revisions![change.current_revision!];
+      return currentGerritPatchset.actions?.['rebase'].enabled === true;
+    }
+    return true;
+  }
+
+  override async isBranchConflicted(
+    baseBranch: string,
+    branch: string
+  ): Promise<boolean> {
+    const searchConfig: GerritFindPRConfig = {
+      state: 'open',
+      branchName: branch,
+      targetBranch: baseBranch,
+    };
+    const change = (await client.findChanges(repository, searchConfig)).pop();
+    if (change) {
+      const mergeInfo = await client.getMergeableInfo(change);
+      return !mergeInfo.mergeable;
+    } else {
+      logger.warn(
+        `There is no open change with branch=${branch} and baseBranch=${baseBranch}`
+      );
+      return true;
+    }
+  }
+
+  override async isBranchModified(branchName: string): Promise<boolean> {
+    const searchConfig: GerritFindPRConfig = { state: 'open', branchName };
+    const change = await client
+      .findChanges(repository, searchConfig, true)
+      .then((res) => res.pop());
+    if (change) {
+      const currentGerritPatchset = change.revisions![change.current_revision!];
+      return currentGerritPatchset.uploader.username !== username;
+    }
+    return false;
+  }
+
+  override async commitAndPush(
+    commit: CommitFilesConfig
+  ): Promise<CommitSha | null> {
+    logger.debug(`commitAndPush(${commit.branchName})`);
+    const searchConfig: GerritFindPRConfig = {
+      state: 'open',
+      branchName: commit.branchName,
+      targetBranch: commit.baseBranch,
+    };
+    const existingChange = await client
+      .findChanges(repository, searchConfig, true)
+      .then((res) => res.pop());
+
+    let hasChanges = true;
+    const origMsg =
+      typeof commit.message === 'string' ? [commit.message] : commit.message;
+    commit.message = [
+      ...origMsg,
+      `Change-Id: ${existingChange?.change_id ?? generateChangeId()}`,
+    ];
+    const commitResult = await git.prepareCommit(commit);
```

Comment:
Another thing we realized is that, within prepareCommit() there is a `git diff HEAD origin/${sourceBranch}`, and on Gerrit this call will always fail because the source branch isn't pushed to the remote.

I think the function should be refactored to:

1. Allow to specify the remote refspec (but keep `origin/${sourceBranch}` by default)
2. Allow to disable such check (and disable it in case it's the first time commiting - this can already be done by setting `commit.force: true`)
3. Pass the correct refspec in case of `existingChange`, like `refs/changes/XX/ABCDEXX/YY` (not sure if such information is available, but if it's not, even always setting `commit.force: true` is better to avoid unnecessary warnings)

Reply:
You are right, the "git diff" you mentioned, between HEAD and origin/$sourceBranch fails always (for Gerrit). 
But the diff checks only for an "empty commit" and i'm doing this by my self in the following lines below with the correct refspec. 
Therefore it should be no problem.

But i can call the prepareCommit like
  `await git.prepareCommit({...commit, force:true});`
to avoid this failing check completely.